src/core/lib/oauth/callback.ts

```diff
@@ -137,6 +137,11 @@ export default async function oAuthCallback(params: {
       })
     }
 
+    // Contains the email, firstName, and lastName of Apple.
+    // Needed beacuse apple does not send the name in the token (profile callback).
+    // profile: {user: {email: 'johndoe@example.com', name: {firstName: 'John', lastName: 'Doe'}}}
+    profile.user = JSON.parse(body?.user ?? query?.user ?? null)
```

Comment:
I really object this change, especially if it's Apple only.

We tested this provider a few days ago, and this wasn't necessary.

Reply:
{
  iss: 'https://appleid.apple.com',
  aud: 'io.ilyeo.Ilyeo.sid',
  exp: 1639202356,
  iat: 1639115956,
  sub: '001195.bb514806f96e4d848b097a33c7f57085.0504',
  at_hash: '2soySvxdgQRZauIl1eZa9w',
  email: 'ryanmalesic@icloud.com',
  email_verified: 'true',
  auth_time: 1639115954,
  nonce_supported: true
}

This is the contents of profile without the changes. Name is not included. Name also isn't stored in the database without the changes I made.

Documentation isn't clear but 
> The modified name is only shared with your app and not with Apple, and hence isn’t included in the ID token.
https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_rest_api/authenticating_users_with_sign_in_with_apple#3383768